src/content/developers/tutorials/short-abi/index.md

```diff
@@ -0,0 +1,632 @@
+---
+title: "Short ABIs for Calldata Optimization"
+description: Optimizing smart contracts for Optimistic Rollups
+author: Ori Pomerantz
+lang: en
+sidebar: true
+tags: ["layer 2", "l2", "optimism", "gas"]
+skill: intermediate
+published: 2022-04-01
+---
+
+## Introduction {#introduction}
+
+In this article you learn about [optimistic rollups](/developers/docs/scaling/optimistic-rollups), the cost of transactions on them, and how that different cost structure requires us to optimize for different things than on the Ethereum mainnet.
+You also learn how to implement this optimization.
+
+### Proper disclosure {#proper-disclosure}
+
+I'm a full time [Optimism](https://www.optimism.io/) employee, so  examples in this article are going to run on Optimism.
+However, the technique explained here should work just as well for other rollups.
+
+### Terminology   {#terminology}
+
+When discussing rollups the term "layer 1" (or L1) is used for mainnet, the production Etherum network.
+The term "layer 2" (or L2) is used for the rollup, or any other system that relies on L1 for security but does most of its processing offchain.
+
+
+## The problem {#the-problem}
+
+### Cost of L2 transactions  {#cost-of-l2-transactions}
+
+[Optimistic rollups](/developers/docs/scaling/optimistic-rollups) have to preserve a record of every historical transaction in such a manner that anybody will be able to go through them and verify that the current state is correct.
+The cheapest way to get data into the Ethereum mainnet is to write it as calldata.
+This solution was chosen by both [Optimism](https://help.optimism.io/hc/en-us/articles/4413163242779-What-is-a-rollup-) and [Arbitrum](https://developer.offchainlabs.com/docs/rollup_basics#intro-to-rollups).
```

Comment:
```suggestion
[Optimistic rollups](/developers/docs/scaling/optimistic-rollups) have to preserve a record of every historical transaction so that anybody will be able to go through them and verify that the current state is correct.
The cheapest way to get data into the Ethereum Mainnet is to write it as calldata.
This solution was chosen by both [Optimism](https://help.optimism.io/hc/en-us/articles/4413163242779-What-is-a-rollup-) and [Arbitrum](https://developer.offchainlabs.com/docs/rollup_basics#intro-to-rollups).
```